src/core/tokenizer.c

```diff
@@ -188,10 +245,13 @@ tokenize(filter_pipeline_t *fp, const char *text, size_t text_len __unused)
 		}
 		tokenset_add(tset, token);
 	}
-	free(content);
+	ubrk_close(it_token);
+	free(utext);
 	return tset;
 err:
```

Comment:
You don't call `ubrk_close()` if there is an error in the loop. You can simplify this by moving `err` label just a little bit up and having one `return` statement at the end. We free the same stuff in both cases.